go/vt/sqlparser/ast_funcs_test.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2022 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sqlparser
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAddQueryHint(t *testing.T) {
+	tcs := []struct {
+		node      Comments
+		queryHint string
+		expected  Comments
+		err       string
+	}{
+		{
+			node:      Comments{},
+			queryHint: "",
+			expected:  Comments{},
+		},
+		{
+			node:      Comments{},
+			queryHint: "SET_VAR(aa)",
+			expected:  Comments{"/*+ SET_VAR(aa) */"},
+		},
+		{
+			node:      Comments{"/* toto */"},
+			queryHint: "SET_VAR(aa)",
+			expected:  Comments{"/*+ SET_VAR(aa) */", "/* toto */"},
+		},
+		{
+			node:      Comments{"/* toto */", "/*+ SET_VAR(bb) */"},
+			queryHint: "SET_VAR(aa)",
+			expected:  Comments{"/*+ SET_VAR(bb) SET_VAR(aa) */", "/* toto */"},
+		},
+		{
+			node:      Comments{"/* toto */", "/*+ SET_VAR(bb) "},
```

Comment:
nit: the parser would never produce a comment like this

Reply:
I maybe wrong, but I personally think that it is better to test the behaviour of a function even for reasonable inputs that it might never receive due to the functioning of a different function